projects/api/src/product/dto/create-product.dto.ts

```diff
@@ -0,0 +1,108 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Transform } from 'class-transformer';
+import { IsNotEmpty, IsNumber, IsPositive, IsString, Matches, MaxLength, Min, MinLength } from 'class-validator';
+
+import { PRODUCT_NAME_VALIDATOR } from '@/utils/constants';
+
+class CreateProductDto {
+  @ApiProperty({
+    type: 'String',
+    description: 'The name of the Product',
+    example: 'Keyboard Red Dragon',
+  })
+  @Matches(PRODUCT_NAME_VALIDATOR, {
+    message: 'The product name must only contain letters, numbers, blank spaces and hyphens.',
+  })
+  @MaxLength(50, {
+    message: 'The product name must be at most 50 characters',
+  })
+  @MinLength(3, {
+    message: 'The product name must be at least 3 characters',
+  })
+  @IsNotEmpty({
+    message: 'The product name must not be empty',
+  })
+  @Transform(({ value }) => value.replace(/\s+/g, ' '))
+  @Transform(({ value }) => value.trim())
+  readonly name: string;
+
+  @ApiProperty({
+    type: 'String',
+    description: 'The description of the product',
+    example:
+      'Experience cable-free freedom and exceptional performance with our Wireless RGB Mechanical Keyboard. Its high-quality switches provide precise tactile responses, while the customizable RGB backlight adds a stylish touch to your workspace. With low-latency wireless connectivity, outstanding durability, and an ergonomic design, this keyboard is the perfect choice for gamers and professionals seeking the ideal balance of form and function.',
+  })
+  @MaxLength(1000, {
```

Comment:
ITS not products "name" its products "description", also change the min and max number of characters on the messsage
@MaxLength(1000, {
Comment
 
Leave a comment
 
    message: 'The **product name** must be at most **50** characters',
  })
  @MinLength(1, {
    message: 'The **product name** must be at least ** 3** characters',
  })
  @IsNotEmpty({
    message: 'The **product name** must not be empty',
  })

Reply:
Fixed